unity/Assets/Scripts/IK_Robot_Arm_Controller.cs

```diff
@@ -517,17 +517,35 @@ public void DropObject()
             jointMeta.rootRelativePosition = FirstJoint.InverseTransformPoint(joint.position);
 
             //local rotation currently relative to immediate parent joint
-            joint.GetChild(0).localRotation.ToAngleAxis(out angleRot, out vectorRot);//getchild to grab the angler since that is what actually changes the geometry angle
-            jointMeta.localRotation = new Vector4(vectorRot.x, vectorRot.y, vectorRot.z, angleRot);
+            if (joint.rotation != joint.GetChild(0).rotation)
```

Comment:
This is not a good way to compare floats, I don't think it would fix the issue, do an epsilon comparison on the components, or a squared vector magnitude comparison

Reply:
What's inherently wrong about comparing two Quaternions against each other?

Reply:
Just to add context: these are quaternions (not floats), and the Quaternion class overrides overrides the `==` comparison such that floating point issues are handled automatically:
https://docs.unity3d.com/540/Documentation/ScriptReference/Quaternion-operator_eq.html

> This function tests whether dot product of two quaternions is **close** to 1.0.

So, this should be fine.